ocs_ci/utility/vsphere.py

```diff
@@ -785,8 +785,19 @@ def check_folder_exists(self, name, cluster, dc):
 
         vms = dc.vmFolder.childEntity
         for vm in vms:
-            if vm.name == name:
-                _rc = True
+            try:
+                if vm.name == name:
+                    _rc = True
+            except vmodl.fault.ManagedObjectNotFound as ex:
+                logger.exception(
+                    "There was an exception hit while attempting to check if folder exists!"
+                )
+                if (
+                    "has already been deleted or has not been completely create"
+                    not in str(ex)
+                ):
+                    raise
```

Comment:
it would be better if we describe which exception to raise here

Reply:
Whole exception details will be described as part of looger.exception as it automatically include exception details.
```
logger.exception(
```

The `raise` is just simple re-raise of exception caught in except block, so nothing to specify here IMO.